### PR TITLE
Release notes for 1.76

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**
Cargo, library stabilizations and some cleanups, particularly to future compat, still pending.

cc @cuviper @rust-lang/release

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -1,3 +1,98 @@
...
+
+Libraries
+---------
+- [adds a column number to `dbg!()`](https://github.com/rust-lang/rust/pull/114962/)
+- [Add `std::hash::{DefaultHasher, RandomState}` exports](https://github.com/rust-lang/rust/pull/115694/)
+- [Fix rounding issue with exponents in fmt](https://github.com/rust-lang/rust/pull/116301/)
+- [Stablize arc_unwrap_or_clone](https://github.com/rust-lang/rust/pull/116949/)
+- [Add T: ?Sized to `RwLockReadGuard` and `RwLockWriteGuard`'s Debug impls.](https://github.com/rust-lang/rust/pull/117138/)
+- [Stabilize `ptr::{from_ref, from_mut}`](https://github.com/rust-lang/rust/pull/117824/)
+- [Stabilize RFC3324 dyn upcasting coercion](https://github.com/rust-lang/rust/pull/118133/)
```
Comment (Urgau) [inline RELEASES.md:35]: This was stabilized by a T-lang [fcp](https://github.com/rust-lang/rust/issues/65991#issuecomment-1670127912); should probably be moved into the Language section.

Comment (slanterns) [reply RELEASES.md:35]: Anyway the stsbilization has been reverted now.

File: RELEASES.md [hunk 2]
```diff
@@ -1,3 +1,102 @@
...
+- [Remove now deprecated target x86_64-sun-solaris.](https://github.com/rust-lang/rust/pull/118091/)
+- [Improve `print_tts`](https://github.com/rust-lang/rust/pull/114571/)
+  This change can break some naive manual parsing of token trees in proc macro
+  code which expect a particular structure after `.to_string()`, rather than just arbitrary Rust code.
+- [Remove asmjs](https://github.com/rust-lang/rust/pull/117338/)
+- [Make `IMPLIED_BOUNDS_ENTAILMENT` into a hard error from a lint](https://github.com/rust-lang/rust/pull/117984/)
+- [Vec's allocation behavior was changed when collecting some iterators](https://github.com/rust-lang/rust/pull/110353)
+  Allocation behavior is currently not specified, nevertheless changes can be surprising.
+  See [`impl FromIterator for Vec`](https://doc.rust-lang.org/nightly/std/vec/struct.Vec.html#impl-FromIterator%3CT%3E-for-Vec%3CT%3E)
+  for more details.
```
Comment (the8472) [inline RELEASES.md:84]: Currently the `FromIterator` impl that I have linked here does not have any documentation.

#120355 will add that on nightly.

**Comment by deltragon** [comment]
Not sure where to best ask this - I was wondering if this should explicitly point out that win7 support is being dropped from the tier 1 windows target, in addition to the new separate win7 target?
For example, in 1.75, the compatibility notes called out the freebsd baseline being raised, this seems like a parallel to that to me.
(Unless I missed something and the tier 1 change was delayed to a later release)

**Comment by Mark-Simulacrum** [comment]
Did a PR dropping that support actually land? The platform support file and quick skim of history doesn't show that we have declared the drop of support. So I think the earliest it will likely happen now is 1.78 (current nightly).

**Comment by slanterns** [comment]
Per https://github.com/rust-lang/compiler-team/issues/651, the decision is to drop the support in 1.76, and we announced it in the [1.76 blog post](https://blog.rust-lang.org/2023/08/24/Rust-1.72.0.html#future-windows-compatibility) (although https://github.com/rust-lang/rust/pull/115141 has not be merged yet.)

**Comment by Mark-Simulacrum** [comment]
We've already built artifacts etc, so I think it'll slip to 1.77 at least, this doesn't seem like something worth rebuilding artifacts for.

**Comment by ChrisDenton** [comment]
Yes, sorry, it's fine to let it slip to 1.77. The intent was to have a blog post out earlier explaining things (and the new win7 target) but I've had a busy new year and not been on top of things. In retrospect it was silly to schedule it so early in the year.

**Pull request merged**